smartsim/_core/mli/infrastructure/worker/worker.py

```diff
@@ -316,20 +334,31 @@ def __init__(self, result: bytes) -> None:
 class RequestBatch:
     """A batch of aggregated inference requests."""
 
-    requests: list[InferenceRequest]
-    """List of InferenceRequests in the batch"""
+    raw_model: t.Optional[Model]
+    """Raw bytes of the model"""
+    callback_descriptors: t.List[str]
+    """The descriptors for channels used for notification of inference completion"""
+    raw_inputs: t.List[t.List[bytes]]
+    """Raw bytes of tensor inputs"""
+    input_meta: t.List[t.List[TensorMeta]]
+    """Metadata about the input data"""
+    input_keys: t.List[t.List[TensorKey]]
+    """A list of tuples containing a (key, descriptor) pair"""
+    output_key_refs: t.Dict[str, t.List[TensorKey]]
+    """A dictionary mapping callbacks descriptors to output keys"""
     inputs: t.Optional[TransformInputResult]
     """Transformed batch of input tensors"""
     model_id: "ModelIdentifier"
     """Model (key, descriptor) tuple"""
 
     @property
-    def has_valid_requests(self) -> bool:
-        """Returns whether the batch contains at least one request.
+    def has_callbacks(self) -> bool:
+        """Determines if the batch has at least one callback channel
+        available for sending results.
 
-        :returns: True if at least one request is available
+        :returns: True if at least one callback is present
         """
-        return len(self.requests) > 0
+        return len(self.callback_descriptors) > 0
```

Comment:
I think the list will always be populated (maybe with `None`) so this test should always return True. Would it make sense to turn it into something like `any(desc is not None for desc in self.callback_descriptors)`?

Reply:
The way I gather the callback_descriptors from the requests is
```py
callback_descriptors=[
      request.callback_desc for request in requests if request.callback_desc
],
```
so there won't be any `None` values and if there are no callbacks in any of the requests it'll be an empty list.

Reply:
Right again